### PR TITLE
added module to apply some ID selection at HLT

### PR DESCRIPTION
**Pull request by folguera**
Added a module inside RecoMuon/MuonIdentification that allows to apply some ID to a collection of reco::muon objects and provides a new collection of reco::muons. It is meant to be used at HLT to apply some mild ID requirements inside the reconstruction sequence. 

It needs backport to 10_0_X.

### CODE REVIEW

**Comment by perrotta** [comment]
please test

File: RecoMuon/MuonIdentification/plugins/MuonIDFilterProducerForHLT.h [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+   explicit MuonIDFilterProducerForHLT(const edm::ParameterSet&);
+   
+   ~MuonIDFilterProducerForHLT() override;
+   
+   void produce(edm::StreamID, edm::Event&, const edm::EventSetup&) const override;
+   static void fillDescriptions(edm::ConfigurationDescriptions & descriptions);
+
+ private:
+   edm::InputTag muonTag_;
+   edm::EDGetTokenT<reco::MuonCollection> muonToken_;
```
Comment (perrotta) [inline RecoMuon/MuonIdentification/plugins/MuonIDFilterProducerForHLT.h]: Private members of global modules should better be all const

File: RecoMuon/MuonIdentification/plugins/MuonIDFilterProducerForHLT.cc [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+   iEvent.getByToken(muonToken_, muons);
+   
+   for ( unsigned int i=0; i<muons->size(); ++i ){
+     const reco::Muon& muon(muons->at(i));
+     if (applyTriggerIdLoose_ && muon::isLooseTriggerMuon(muon)) { 
+       output->push_back(muon);
+     }
+     else {  // Implement here manually all the required/desired cuts 
+
+
```
Comment (perrotta) [inline RecoMuon/MuonIdentification/plugins/MuonIDFilterProducerForHLT.cc]: (If you are going to update this PR) you can remove one on these empty lines

File: RecoMuon/MuonIdentification/plugins/MuonIDFilterProducerForHLT.cc [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+       output->push_back(muon);
+     }
+     else {  // Implement here manually all the required/desired cuts 
+
+       
+       if ( (muon.type() & allowedTypeMask_) == 0 )                  continue;
+       if ( (muon.type() & requiredTypeMask_) != requiredTypeMask_ ) continue;
+       // tracker cuts
+       if ( !muon.innerTrack().isNull() ){
+	 if (muon.innerTrack()->hitPattern().trackerLayersWithMeasurement() < min_NTrkLayers_) continue;
```
Comment (perrotta) [inline RecoMuon/MuonIdentification/plugins/MuonIDFilterProducerForHLT.cc]: (If you are going to update this PR) the same two spaces indentation that you use everywhere else would make this class better readable, here and below

**Comment by perrotta** [comment]
please test

**Comment by perrotta** [comment]
+1
- A new filter is just defined to be used eventually at HLT
- It doesn't enter any reco (or even HLT) sequence by now: no effect whatsoever on reco outputs

**Comment by fabiocos** [comment]
+1

**Pull request merged**